common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -293,7 +295,9 @@ case class Activation(name: String,
                       conductor: Boolean,
                       memory: Int,
                       causedBy: Option[String],
-                      size: Option[Int] = None)
+                      size: Option[Int] = None,
+                      actionStatusCode: Option[Int] = None
```

Comment:
The inspection of the result of an action (instead of treating it opaquely) is a down side to this approach where otherwise the result is completely opaque. Naturally it would only work for actions that actually return an http response with a status code, which I don't think you can distinguish at the point of constructing a `Message`. Did you consider capturing the HTTP status code from the edge router instead for a /api/v1/web actions?

Reply:
> The inspection of the result of an action (instead of treating it opaquely) is a down side to this approach where otherwise the result is completely opaque

The `statusCode` field is a documented structure for [web actions][1]. `statusCode` handling when under `error` field though is not documented properly. So current logic only looks for these documented structure to extract the info

> Did you consider capturing the HTTP status code from the edge router instead for a /api/v1/web actions

@rabbah If we capture the metrics at edge router then its tricky to distinguish a 5xx status from system error or web action actually responding with 5xx. Hence need to capture it at activation response level.

[1]: https://github.com/apache/openwhisk/blob/master/docs/webactions.md#handling-http-requests-with-actions